File: backend/src/hono/routes/retry_policy/retry_policy_test.ts
import { describe, it, expect } from 'vitest';
import retryPolicy from './retry_policy';

describe('retryPolicy', () => {
  it('retorna true para timeout', async () => {
    expect(await retryPolicy(null, 'timeout')).toBe(true);
  });

  it('retorna true para connection_error', async () => {
    expect(await retryPolicy(null, 'connection_error')).toBe(true);
  });

  it('retorna false para outros erros', async () => {
    expect(await retryPolicy(null, 'outro_erro')).toBe(false);
  });
});

File: lib/features/retry_policy/retry_policy_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:passdriver/features/retry_policy/retry_policy.dart';

void main() {
  test('shouldRetry retorna true para timeout', () {
    final retryPolicy = RetryPolicy();
    expect(retryPolicy.shouldRetry('timeout'), true);
  });

  test('shouldRetry retorna true para connection_error', () {
    final retryPolicy = RetryPolicy();
    expect(retryPolicy.shouldRetry('connection_error'), true);
  });

  test('shouldRetry retorna false para outros erros', () {
    final retryPolicy = RetryPolicy();
    expect(retryPolicy.shouldRetry('outro_erro'), false);
  });
}
